web/packages/teleterm/src/mainProcess/windowsManager.ts

```diff
@@ -272,6 +272,43 @@ export class WindowsManager {
     app.focus({ steal: true });
   }
 
+  /**
+   * Returns a promise that resolves after window is focused or after a timeout, or after the
+   * passed signal is aborted. There's no guarantee that the window receives focus, hence the
+   * built-in timeout.
+   */
+  waitForWindowFocus(signal?: AbortSignal, timeoutMs?: number): Promise<void> {
+    if (signal?.aborted) {
+      return Promise.resolve();
+    }
+
+    return new Promise(resolve => {
+      if (signal?.aborted) {
+        resolve();
+        return;
+      }
+
+      const interval = setInterval(() => {
+        if (this.window.isFocused()) {
+          resolve();
+          clearInterval(interval);
+          clearTimeout(timeout);
+        }
+      }, 16);
+
+      const timeout = setTimeout(() => {
+        resolve();
+        clearInterval(interval);
+      }, timeoutMs);
```

Comment:
If I'm correct, we don't set the timeout in production code. How does this work then? `setTimeout` will resolve the promise before we even check the interval. Shouldn't we have a non-zero `timeoutMs`?

Reply:
Oops, I swear I had a default set to 400ms and then it got lost when I was writing a test (where I wanted to use a shorter timeout).

Reply:
I could swear that I tested it while it had no timeout and it worked. I retested it again and it indeed does work. But I suspect it's just a matter of this being a race condition – that extra tick gives the OS just enough time to focus the window before the rest of the logic attempts to use the clipboard API.